crates/libs/windows/src/core/factory_cache.rs

```diff
@@ -44,11 +49,20 @@ impl<C: RuntimeName, I: Interface> FactoryCache<C, I> {
     }
 }
 
-/// Attempts to load the factory interface for the given WinRT class.
-pub fn factory<C: RuntimeName, I: Interface>() -> Result<I> {
+/// Attempts to load the factory interface for the given WinRT class, and library
+/// if provided.
+pub fn factory<C: RuntimeName, I: Interface>(library: Option<&str>) -> Result<I> {
```

Comment:
Perhaps split this into two functions: `factory` unchanged with no params and then `factory_from_library` that takes a library.